packages/ember-views/lib/views/each.js

```diff
@@ -53,17 +53,38 @@ export default CollectionView.extend(_Metamorph, {
   },
 
   _assertArrayLike(content) {
+    var theArrayStr;
+    var shouldShowError;
+    var hasDetectedArray;
+
     Ember.assert(fmt("The value that #each loops over must be an Array. You " +
                      "passed %@, but it should have been an ArrayController",
                      [content.constructor]),
                      !ControllerMixin.detect(content) ||
                        (content && content.isGenerated) ||
                        content instanceof ArrayController);
+
+    hasDetectedArray = EmberArray.detect(content);
+    if (ControllerMixin.detect(content) && content.get('model') !== undefined) {
+      theArrayStr = fmt("'%@' (wrapped in %@)", [content.get('model'), content]);
+      shouldShowError = !Ember.EXTEND_PROTOTYPES &&
+                        Array.isArray(content.get("model")) &&
```

Comment:
Array.isArray isn't IE8 safe. We have a helper called `isArray` in ember-metal/lib/utils i think.